pkg/handlers/registry.go

```diff
@@ -210,21 +210,31 @@ func (h *Handler) UpdateAppRegistry(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	skipImagePush := updateAppRegistryRequest.IsReadOnly
+	if foundApp.IsAirgap {
+		// TODO: pushing images not yet supported in airgapped instances.
+		skipImagePush = true
+	}
+
+	latestSequence, err := store.GetStore().GetLatestAppSequence(foundApp.ID, true)
+	if err != nil {
+		logger.Error(errors.Wrapf(err, "failed to get latest app sequence for app %s", foundApp.Slug))
+		updateAppRegistryResponse.Error = err.Error()
+		JSON(w, http.StatusInternalServerError, updateAppRegistryResponse)
+		return
+	}
```

Comment:
Moved this bit out of the goroutine since we don't want to set an initial task status of `"running"` if the query fails.